src/addons/createSingleton.ts

```diff
@@ -111,6 +118,51 @@ const createSingleton: CreateSingleton = (
     triggerTarget: references,
   }) as CreateSingletonInstance<CreateSingletonProps>;
 
+  const originalShow = singleton.show;
+
+  singleton.show = function(target?: ReferenceElement|Instance|number):void {
+    originalShow();
+
+    // first time, showOnCreate or programmatic call with no params
+    // default to showing first instance
+    if (!currentTarget && target == null) {
+      return prepareInstance(singleton, references[0]);
+    }
+
+    // triggered from event (do nothing as prepareInstance already called by onTrigger)
+    // programmatic call with no params when already visible (do nothing again)
+    if (currentTarget && target == null) {
+      return
+    }
+
+    // target is index of instance
+    if (typeof target === 'number') {
+      return references[target] && prepareInstance(singleton, references[target]);
+    }
+
+    // target is specific Instance or Element
+    return prepareInstance(singleton, (target as Instance)?.reference || target);
+  }
+
+  singleton.showNext = function():void {
+    const first = references[0];
+    if (!currentTarget) {
+      return singleton.show(0);
+    }
+    const index = references.indexOf(currentTarget);
+    singleton.show(references[index+1] || first);
```

Comment:
Have you got prettier running? There should be a space between the number and + (`[index + 1]`)

Reply:
done